src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/ClusterMonitorActor.scala

```diff
@@ -169,6 +169,10 @@ class ClusterMonitorActor(val cluster: Cluster,
       if (shouldRecreateCluster(errorDetails.code, errorDetails.message)) {
         // Update the database record to Deleting, shutdown this actor, and register a callback message
         // to the supervisor telling it to recreate the cluster.
+        cluster.stagingBucket match {
```

Comment:
Not even sure we should be deleting the staging bucket here -- if cluster creation is failing and we're trying it would be nice to keep a record of the failing logs.

Reply:
+1 leave them there -- part of this is to aid debugging!